src/main/java/mate/academy/bookdao/BookDaoImpl.java

```diff
@@ -0,0 +1,145 @@
+package mate.academy.bookdao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dbconnection.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public int getRowsCount() {
+        String sql = "SELECT COUNT(*) AS total_rows FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        PreparedStatement preparedStatement = connection
+                                .prepareStatement(sql);
+                        ResultSet resultSet = preparedStatement.executeQuery()) {
+            if (resultSet.next()) {
+                return resultSet.getInt("total_rows");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get row count", e);
+        }
+        return 0;
+    }
+
+    @Override
+    public boolean clear() {
+        String sql = "TRUNCATE TABLE books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                        Statement statement = connection.createStatement()) {
```

Comment:
Use `PreparedStatement` over `Statement`, even for a static query with no parameters in `clear()` method. It's the best practice, and it's slightly faster.